### PR TITLE
[develop] Trello-XVMQSa0c: Fix authentication flow

### PR DESCRIPTION
**Pull request by wildeastengineer**
[Trello-XVMQSa0c](https://trello.com/c/XVMQSa0c): Fix authentication flow

### CODE REVIEW

File: client/repositories/cookie/CookieServer.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+    save(name, value, options = {}) {
+        this.res.cookie(name, value, options);
+    }
+
+    remove(name, options = {}) {
+        this.res.clearCookie(name, options);
+    }
+}
```
Comment (AndreySurzhan) [inline client/repositories/cookie/CookieServer.js]: semicolon

File: client/repositories/cookie/CookieClient.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        document.cookie = updatedCookie;
+    }
+
+    remove(name) {
+        this.save(name, '', {
+            expires: -1
+        });
+    }
+}
```
Comment (AndreySurzhan) [inline client/repositories/cookie/CookieClient.js]: semicolon

File: client/repositories/cookie/Cookie.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    save(name, value, options = {}) {
+        return this.cookie.save(name, value, options);
+    }
+
+    remove(name) {
+        return this.cookie.remove(name);
+    }
+}
```
Comment (AndreySurzhan) [inline client/repositories/cookie/Cookie.js]: semicolon

File: client/routes/index.js [hunk 4]
```diff
@@ -1,3 +1,7 @@
-import routes from './routes';
+import getRoutes from './routes';
+import paths from './paths';
 
-export default routes;
\ No newline at end of file
+export {
+    getRoutes,
+    paths
+};
```
Comment (AndreySurzhan) [inline client/routes/index.js]: extra line

File: client/routes/routes.jsx [hunk 5]
```diff
@@ -1,18 +1,32 @@
...
+import { Page404 } from 'components/pages/ErrorPages'
+
+export default function (store) {
+    function onEnter(nextState, replace) {
+        const { getState } = store;
+
+        if (!isAuthenticated(getState())) {
+            replace(`/?redirect=${nextState.location.pathname}`);
+        }
+    }
```
Comment (AndreySurzhan) [inline client/routes/routes.jsx]: semicolon

Comment (wildeastengineer) [inline client/routes/routes.jsx]: @AndreySurzhan semicolons are not necessary after `function` and `class` declarations

**Pull request merged**